### ISSUE TITLE
Illegal instruction (core dumped) using tensorflow==2.4.0 with AVX instruction when importing

### ISSUE THREAD

**Issue by H-Li24** [tensorflow/tensorflow#46108]
**System information**
- OS Platform and Distribution: Linux Ubuntu 18.04)
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.4.0
- Python version: 3.7.9
- CUDA/cuDNN version: 11.0/8.0.5
- GPU model and memory: GTX 2080Ti

**Describe the current behavior**

Attempting to import tensorflow produces an "Illegal instruction(core dumped)" error.

**Describe the expected behavior**

Import tensorflow without error.

**Standalone code to reproduce the issue**
```
conda create -n test-tf2.4 python=3.7
conda activate test-tf2.4
pip install tensorflow==2.4.0
```
The following command exits with an "Illegal instruction" error:
```
python -c "import tensorflow as tf"
```

**Other info / logs** 
This will be not an issue with the last development version of 2.5.0 (tf-nightly 2.5.0.dev20210102) from tf-nightly, installed with:
```
pip install tf-nightly
```

MOST IMPORTANTLY, My machine has AVX instructions. So this is not the same as the previous "old cpu", while it is THE SAME AS the [#44668 ](https://github.com/tensorflow/tensorflow/issues/44668)problem  which your team has NOT SOLVED.

**Comment by otoomet** [comment]
I can confirm the problem.  I have i7-3770 which supports AVX (but not AVX2) and gt1030.  
 
    python -m tensorflow

(tf 2.4.0 installed with pip in venv) crashes with "illegal instruction" while the same works when installing tf-nightly.  Otherwise running in ubuntu 20.04, python 3.8.5.  Nvidia-smi reports cuda 11 while nvcc reports cuda 10.1, so there seems to be some sort of mismatch.

gdb suggests the issue happens with `vcvtsi2sd` instruction in the middle row below (but I know too little of gdb and assembly to actually tell).

    KSt8functionIFNS_6StatusERKNS_7VariantEPS7_S5_IFS6_RKNS_6TensorEPSB_EEEE+520>    shlx   %r14d,%edx,%eax
    KSt8functionIFNS_6StatusERKNS_7VariantEPS7_S5_IFS6_RKNS_6TensorEPSB_EEEE+525>    vcvtsi2sd %rax,%xmm0,%xmm0
    KSt8functionIFNS_6StatusERKNS_7VariantEPS7_S5_IFS6_RKNS_6TensorEPSB_EEEE+530>    vmulsd %xmm2,%xmm0,%xmm0

**Comment by mihaimaruseac** [comment]
Duplicate of #45744 #45866 and a few more. We already have a fix, cherrypicked to the 2.4 branch and waiting for patch release.
